File: examserver/src/main/java/com/exam/service/QuizService.java
package com.exam.service;

import com.exam.model.Category;
import com.exam.model.Quiz;

import java.util.Set;

public interface QuizService {
    Quiz addQuiz(Quiz quiz);
    Quiz updateQuiz(Quiz quiz);
    Set<Quiz> getQuiz();
    Quiz getQuiz(Long id);
    void deleteQuiz(Long id);
    Set<Quiz> findByCategoryId(Long id);
    Set<Quiz> findByActive(Boolean active);
    Set<Quiz> findByCategoryAndActive(Long id,Boolean active);

}

File: examfront/src/app/services/quiz.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baseUrl from './helper';

@Injectable({
  providedIn: 'root'
})
export class QuizService {

  constructor(private http:HttpClient) { }

  getquiz()
  {
    return this.http.get(`${baseUrl}/quiz`);
  }
  addquiz(data:any)
  {
    return this.http.post(`${baseUrl}/quiz`,data);
  }
  deletequiz(id:number)
  {
    return this.http.delete(`${baseUrl}/quiz/${id}`);
  }

  getSinglequiz(id:any)
  {
    return this.http.get(`${baseUrl}/quiz/${id}`)
  }
  updateQuiz(data:any)
  {
    return this.http.put(`${baseUrl}/quiz`,data);
  }
  findByCategory(id:any)
  {
    return this.http.get(`${baseUrl}/quiz/category/${id}`)
  }


  getActiveQuiz()
  {
    return this.http.get(`${baseUrl}/quiz/active`);
  }
  getQuizByCategoryAndActive(id:any)
  {
    return this.http.get(`${baseUrl}/quiz/category/${id}/active`)
  }

}
